src/Mimi.jl

```diff
@@ -571,6 +599,50 @@ function run_timestep(s::adder, t::Int)
     v.output[t] = p.input[t] + p.add[t]
 end
 
+#Begin Graph Functionality section
+
+function show(io::IO, m::Model)
+    println(io, "showing model component connections:")
+    i=1
+    for c in keys(m.components2)
+        i_connections = get_connections(m,c,:incoming)
+        o_connections = get_connections(m,c,:outgoing)
+        println(io, i, ". ", c, " component")
+        println(io, "    incoming parameters:")
+        if length(i_connections)==0
+            println(io, "      none")
+        else
+            [println(io, "      - ",e.target_parameter_name," from ",e.source_component_name," component") for e in i_connections]
+        end
+        println(io, "    outgoing variables:")
+        if length(o_connections)==0
+            println(io, "      none")
+        else
+            [println(io, "      - ",e.source_variable_name," in ",e.target_component_name, " component") for e in o_connections]
+        end
+        i += 1
+    end
+end
+
+function get_connections(m::Model, c::ComponentInstanceInfo, which::Symbol)
+    return get_connections(m, c.name, which)
+end
+
+function get_connections(m::Model, component_name::Symbol, which::Symbol)
+    if which==:all
+        bool(e) = e.source_component_name==component_name || e.target_component_name==component_name
```

Comment:
For some reason this causes troubles on julia 0.5. I don't understand why, but the following should solve the problem: instead of defining a regular function here, create an anonymous function and assign it to a variable. Maybe, just to be save, also don't use ``bool``, that sounds very much like something that julia might use at some point. This line would look like this:
````julia
f = e-> e.source_component_name==component_name || e.target_component_name==component_name
````
And then you pass ``f`` to ``filter``.